lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2714,7 +2719,9 @@ class HSADevice final : public KalmarDevice
         } co_data;
 
         status = hsa_isa_from_name(isa.c_str(), &co_data.isa_type);
-        STATUS_CHECK(status, __LINE__);
+
+	if (status != HSA_STATUS_SUCCESS)
+		return false;
```

Comment:
please fix indentation

Reply:
Also `kernel_source` needs to be freed before returning or turn that buffer into an automatic object

Reply:
done